### DISCUSSION TITLE
Does Tensor Parallelism support current scaling mode fp8 ?

### DISCUSSION THREAD

**Discussion by kuozhang** [hpcaitech/ColossalAI#6101] [Community | Q&A] [answered]
Hello, I checked the code about fp8 (not fp8 communication) and found how it works is to rewrite_op F.linear, but  I didn't found out whether Linear1D_Col and Linear1D_Row support fp8, since if use fp8 of current scaling mode, the amax values of sharded weight and activations are needed, and this needs an communication of allreduce(reduceop=max)。

**Comment by Edenzzzz** [comment]
Indeed this seems unimplemented for now. You're welcome to submit a PR or ping other members.

**Comment by Edenzzzz** [comment, marked as answer]
It seems that Transformers Engine's original implementation doesn't do this. I opened an issue #6105 for discussion.

**Comment by xXMrNidaXx** [comment]
FP8 with Tensor Parallelism in ColossalAI:

**Current support:**
FP8 is relatively new and TP support is evolving.

**Check your setup:**
```python
import torch
print(torch.cuda.get_device_capability())  # Need 8.9+ for native FP8
# H100, RTX 4090 support FP8
```

**ColossalAI FP8 config:**
```python
from colossalai.booster import Booster
from colossalai.booster.plugin import HybridParallelPlugin

plugin = HybridParallelPlugin(
    tp_size=2,
    precision="fp8",  # If supported
    # or
    fp8_communication=True
)
```

**Current limitations:**
- Not all ops have FP8 kernels
- Communication often still in FP16/BF16
- Model-specific support

**Alternatives:**
```python
# Mixed precision with scaling
plugin = HybridParallelPlugin(
    tp_size=2,
    precision="bf16",
    enable_gradient_scaling=True
)
```

**For now:**
- Use BF16 for stable TP
- FP8 for compute, FP16 for communication
- Check latest releases for updates

We train large models at [RevolutionAI](https://revolutionai.io). What GPU and model size are you working with?
